march_hardware/src/error/motion_error.cpp

```diff
@@ -63,5 +73,19 @@ std::string parseDetailedError(uint16_t detailed_error)
   }
   return description;
 }
+
+std::string parseSecondDetailedError(uint16_t second_detailed_error)
+{
+  std::string description;
+  const std::bitset<16> bitset(second_detailed_error);
+  for (size_t i = 0; i < SECOND_DETAILED_MOTION_ERROR_SIZE; i++)
+  {
+    if (bitset.test(i))
+    {
+      description += SECOND_DETAILED_MOTION_ERRORS[i];
+    }
+  }
+  return description;
+}
```

Comment:
It would be nice to generalize this parsing of errors. Now we have 3 duplicates, which is hard to maintain.

Reply:
Fair, let me see what I can do